authLdap.php

```diff
@@ -455,6 +464,18 @@ function authLdap_groupmap($username, $dn)
     }
 
     try {
+        // To allow searches based on LDAP uid instead of the one used in Filter input, we replace the
+        // string %uid% with the users UID.
+        $authLDAPGroupFilter = str_replace('%uid%', $realuid, $authLDAPGroupFilter);
+        authLdap_debug('Group Filter: ' . json_encode($authLDAPGroupFilter));
+        $groups = authLdap_get_server()->search(sprintf($authLDAPGroupFilter, $username), array($authLDAPGroupAttr));
+    } catch(Exception $e) {
+        authLdap_debug('Exception getting LDAP group attributes: ' . $e->getMessage());
+        return '';
+    }
+
+
+    try {
```

Comment:
this should not be necessary as the ```%s```-string will be replaced by the ```$username```-parameter which contains the ```realuid```-value